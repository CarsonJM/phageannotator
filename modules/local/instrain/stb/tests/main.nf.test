nextflow_process {

    name "Test Process: INSTRAIN_STB"
    script "../main.nf"
    process "INSTRAIN_STB"
    tag "modules"
    tag "modules_local"
    tag "instrain/stb"


    test("['homo_sapiens']['genome']['genome_fasta']") {

        when {
            process {
                """
                input[0] = [ [ id: 'reference_fasta' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
