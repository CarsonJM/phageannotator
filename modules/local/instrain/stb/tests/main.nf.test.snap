{
    "['homo_sapiens']['genome']['genome_fasta']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,8f9e23730f7fda6be33ce21216fb7f2d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ],
                "stb": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,8f9e23730f7fda6be33ce21216fb7f2d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ]
            }
        ],
        "timestamp": "2023-12-04T16:35:57.01447565"
    },
    "['homo_sapiens']['genome']['genome_fasta'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ],
                "stb": [
                    [
                        {
                            "id": "reference_fasta"
                        },
                        "reference_fasta.stb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7df22a05d168d97527754f453fe59894"
                ]
            }
        ],
        "timestamp": "2024-01-17T22:32:54.38815316"
    }
}